Fixed performance issues for deny permission checks or `runWithoutAuthorization` blocks.

theo-ritense opened the pull request:
<!-- Please consider the following points before creating a pull request. -->

### Describe the changes
Link to the related Github issue: https://github.com/valtimo-platform/valtimo-issues/issues/53

<!-- Please add any additional comments that might be relevant for reviewing this pull request  -->
<!-- Why did you choose to make these changes? Were there any trade-offs you had to consider?   -->
<!-- Note: add an empty line with a > to use multiple lines  -->
**Relevant comments:**
> 

### Breaking changes
<!-- Valtimo aims to comply with the SemVer specification.  -->
<!-- Breaking changes are only allowed in the `next-major` branch.  -->
- [ ] The contribution only contains changes that are not breaking.

### Documentation
<!-- Release notes should be available in the Valtimo documentation.  -->
- [ ] Release notes have been written for these changes. 

Link to the pull request in the [Valtimo documentation repository](https://github.com/valtimo-platform/valtimo-documentation):
>

New features or changes that have been introduced have been documented.
- [ ] Yes
- [ ] Not applicable

### Tests
Unit tests have been added that cover these changes
- [ ] Yes
- [ ] Not applicable

Integration tests have been added that cover these changes
- [ ] Yes
- [ ] Not applicable

### Security
The [Secure by Design principle](https://en.wikipedia.org/wiki/Secure_by_design) has been applied to these changes
- [ ] Yes
- [ ] Not applicable

Added or changed REST API endpoints have authentication and authorization in place
- [ ] Yes
- [ ] Not applicable

Valtimo access control checks have been implemented
- [ ] Yes
- [ ] Not applicable

### Dependencies
Newly added dependencies do not introduce known vulnerabilities/CVE's and are in line with the Valtimo license
- [ ] Yes
- [ ] Not applicable

theo-ritense committed:
Fixed performance issues for deny permission checks or `runWithoutAuthorization` blocks.

Also fixed a performance issue at form prefilling where associated processes were checked even when `processVarPointers` was empty.

marijnritense commented on authorization/src/main/kotlin/com/ritense/authorization/specification/AuthorizationSpecification.kt:
```diff
@@ -33,11 +33,17 @@ import jakarta.persistence.criteria.Predicate
-    protected val permissions: List<Permission>
+    protected val permissionSupplier: Supplier<List<Permission>>
 ) : Specification<T> {
+    @Deprecated("Use the constructor with permission supplier instead")
```
Can you change it to this format?

`@Deprecated("Since 12.1.0", ReplaceWith("com.ritense.search.service.SearchFieldV2Service.delete()"))`

marijnritense commented on authorization/src/main/kotlin/com/ritense/authorization/specification/AuthorizationSpecificationFactory.kt:
```diff
@@ -18,11 +18,23 @@ package com.ritense.authorization.specification
-    fun canCreate(request: AuthorizationRequest<*>, permissions: List<Permission>): Boolean
+    fun canCreate(request: AuthorizationRequest<*>, permissionSupplier: Supplier<List<Permission>>): Boolean = canCreate(request, permissionSupplier.get())
+
+    @Deprecated("Use the create method with a permission supplier instead")
```
Same here

marijnritense commented on authorization/src/main/kotlin/com/ritense/authorization/specification/AuthorizationSpecificationFactory.kt:
```diff
@@ -18,11 +18,23 @@ package com.ritense.authorization.specification
+    }
+
+    // Change this to something more dynamic in the future
+    @Deprecated("Use the canCreate method with a permission supplier instead")
```
Same here

theo-ritense committed:
- PR feedback
- Had to let URLProcessLinkService use the ValtimoAuthorizationService interface to make mockito play nice again.

theo-ritense committed:
Moved log message building into logging lambda.
